hicommonwealth/commonwealth
# New Subscription Model Migration

## Link to Issue
Closes: #6916

## Description of Changes
Created a migration which does the following:
- Creates all of the tables (with indexes) as described in the issue
- Migrates existing data from the Subscriptions table into the new tables
- Sets up triggers that insert into and delete from the new tables to ensure they remain in sync with the Subscriptions table

Created all the entity schemas + Sequelize models for the new tables.

## Test Plan
- Run the migration
- Start the server
- Modify the subscriptions like normal
- Ensure the subscription changes made it into the new tables

## Deployment Plan

1. 

## Other Considerations

- The migration will take quite a while to run (10s on Frick) but it will not lock the DB since all the queries are either on new unutilized tables or SELECT queries on existing tables.
- CommunityAlerts covers both `chain-event` and `snapshot-proposal` subscriptions which can both be created/destroyed and enabled/disabled. Due to this, a CommunityAlerts subscription will only exist if both the chain-event AND snapshot subscriptions exist and are enabled (if a community has both chain-events and a snapshot space - if they have only one or the other, subscriptions old and new will be perfectly in sync).
